tools/java/org/unicode/cldr/api/CldrDataSupplier.java

```diff
@@ -237,14 +250,14 @@ public CldrData getDataForType(CldrDataType type) {
 
         // This is a separate function because stream functions cannot throw checked exceptions.
         //
-        // Note: "Files.list()" warns about closing resources and suggests "try-with-resources" to
+        // Note: "Files.walk()" warns about closing resources and suggests "try-with-resources" to
         // ensure closure, "flatMap()" (which is what calls this method) is defined to call close()
         // on each stream as it's added into the result, so in normal use this should all be fine.
         //
         // https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html#flatMap-java.util.function.Function-
         private static Stream<Path> listXmlFiles(Path dir) {
             try {
-                return Files.list(dir).filter(IS_XML_FILE);
+                return Files.walk(dir).filter(IS_XML_FILE);
             } catch (IOException e) {
                 throw new RuntimeException(e);
```

Comment:
> This should be new ICUUncheckedIOException(e), not a raw RuntimeException.
Interesting. There will be many more cases I suspect.
Given that this is completely un-recoverable I'm wondering what the benefit of using ICUUncheckedIOException is and what called could possibly do differently in the face of this compared to a RuntimeException (so I can document it in the code).

Reply:
Actually, given that the API is already Java 8 and that exposing ICUUncheckedIOException means leaking CLDRFile into the API (it's an inner class) or wrapping it, what about just throwing UncheckedIOException since that's a thing now.

Reply:
I changed to UncheckIOException (Java8) and re-uploaded. Should be good for review again.